Kernel/include/semaphores.h

```diff
@@ -0,0 +1,78 @@
+#ifndef _SEMAPHORE_H_
+#define _SEMAPHORE_H_
+
+#include <stddef.h>
+#include <lib.h>
+#include <memoryManager.h>
+#include <resourceNamerADT.h>
+#include <waitQueueADT.h>
+#include <kernelTypes.h>
+#include <scheduler.h>
```

Comment:
The only includes that seem to be needed in the .h are `stddef.h` and `kernelTypes.h`, move the rest to the .c if needed.